app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -1189,18 +1189,15 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope {
                     val site = viewModel.siteLiveData.value
                     val events = site?.trackingEvents
 
-                    val act = activity
-                    if (act != null) {
+                    activity?.let { activity ->
```

Comment:
I did a quick look to the previous merged PR and to this one.

I would like to share some thoughts here:
1. We should not reach this point if the activity is null, rather than doing a null check. I think the fix will be just to cancel the coroutine. From what I see, delay is suspending this block, if user leaves the screen and we don't cancel the coroutine, the coroutine will be resumed and that's why the crash will happen for `requireActivity()`.
2. Not canceling also means that we are leaking.

Suggestion: as a quick fix, without checking all the different coroutines that we are starting from `BrowserTabFragment`, I would keep the reference to the `SupervisorJob`, and cancel `onDestroy`. I think there are some CoroutineContext more aware of the lifecycle, but we can investigate that part in future tasks.

Reply:
I like this idea! In theory, if doing so, we should assume code after `delay` won't run at all in that scenario, since `delay` checks if the coroutine is active for us IIRC

Reply:
That's what I expect. Any method from `kotlinx.coroutines` is aware of cancelations, so code after `delay` should not be executed if coroutine is canceled.

Reply:
This makes sense to me, I have pushed a new commit which includes @cmonfortep's change.